coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -284,7 +284,10 @@ class SingleClusterPlanner(dsRef: DatasetRef,
       series.plans.foreach(_.addRangeVectorTransformer(PeriodicSamplesMapper(newStartMs, lp.stepMs,
         lp.endMs, window, Some(execRangeFn), qContext, lp.stepMultipleNotationUsed,
         paramsExec, lp.offsetMs, rawSource)))
-      series
+      if (execRangeFn == InternalRangeFunction.AbsentOverTime)
+        addAbsentFunctionMapper(series, lp.columnFilters,
+        RangeParams(lp.startMs / 1000, lp.stepMs / 1000, lp.endMs / 1000), qContext)
+      else series
```

Comment:
There is a leak in abstraction. There is some hidden magic (translation of absentOverTime to last sample) that is happening during execution, that this planner depends on. It is hard for reader concerned with planning to understand whats going on without digging very deep into implementation. 

Instead, we should plan with LastSample and apply the mapper right here so that the "magic" is visible in one place. 

Please also add a short description on the secret sauce of range function + existing mapper to achieve absent over time.